tests/integration/views/test_request.py

```diff
@@ -1149,6 +1149,32 @@ def test_requests_for_workspace(airlock_client):
     assert author2.username in response.rendered_content
 
 
+def test_requests_for_workspace_filter(airlock_client, mock_old_api):
+    airlock_client.login(workspaces=["test1"])
+    author2 = factories.create_airlock_user("author2", ["test1"], False)
+
+    factories.create_request_at_status(
+        "test1",
+        author=author2,
+        status=RequestStatus.APPROVED,
+        files=[factories.request_file(approved=True)],
+    )
+
+    factories.create_request_at_status(
+        "test1",
+        author=author2,
+        status=RequestStatus.PENDING,
+    )
+
+    response = airlock_client.get("/requests/workspace/test1?status=APPROVED")
+
+    response.render()
```

Comment:
You shouldn't need to call `render()` explicitly if you're testing with `rendered_content`.

I'd suggest testing the context instead of the string content; i.e. test that `response.context_data["requests_for_workspace"]` contains the requests you expect - that'll be more likely to continue to pass if we change some of the template format later. At the moment you're asserting that "APPROVED" is in the response content, but I think it always will be. If the filter didn't work and returned all requests, including the pending one, this would still match. If it returned no requests, I think it would also still match, because "APPROVED" would be found in the dropdown options.